source/scripts/ContentScript/index.ts

```diff
@@ -79,3 +76,33 @@ const inject = (content: string) => {
 
 inject(`window.SUPPORTED_WALLET_METHODS = ${JSON.stringify(DAppMethods)}`);
 inject(_inject);
+
+const setDappNetworkProvider = (networkVersion?: any, chainId?: any) => {
+  if (networkVersion && chainId) {
+    inject(`window.ethereum.chainId = ${chainId}`);
+    inject(`window.ethereum.networkVersion = ${networkVersion}`);
+
+    return;
+  }
+
+  const ethChainId = web3Provider.send('eth_chainId', []);
+  const ethNetworkVersion = web3Provider.send('net_version', []);
+
+  Promise.all([ethChainId, ethNetworkVersion]).then(([id, version]) => {
+    console.log({ id, version });
+    inject(`window.ethereum.chainId = ${String(id)}`);
+    inject(`window.ethereum.networkVersion = ${version}`);
+  });
+};
+
+browser.runtime.onMessage.addListener(({ type, data }) => {
+  if (type === 'CHAIN_CHANGED')
+    setDappNetworkProvider(data.networkVersion, data.chainId);
+});
+
+// Every message from pali emits an event
+backgroundPort.onMessage.addListener(({ id, data }) => {
+  emitter.emit(id, data);
+});
+
+setDappNetworkProvider();
```

Comment:
This as the above-pointed problem will run at each tab page refresh, causing the initial state of the wallet to be overwritten for syscoin chainId and networkId at each refresh of a tab that has window.ethereum component injected into it. It needs either to be put in somewhere else that just run at the wallet initialization or be able to fetch the current active network from the background